frame/contracts/src/wasm/prepare.rs

```diff
@@ -414,14 +368,15 @@ where
 		// This is not our only defense: We check for float types later in the preparation
 		// process. Additionally, all instructions explicitly  need to have weights assigned
 		// or the deployment will fail. We have none assigned for float instructions.
-		deterministic_only: matches!(determinism, Determinism::Deterministic),
+		floats: matches!(determinism, Determinism::AllowIndeterminism),
```

Comment:
Not a big fan of the negated identifier. Why not `Determinism::{Enable, Disable}` and use `matches!(determinism, Determinism::Disable)` here?

Reply:
The comment attached should be updated, I guess ("*This is not our only defence*" seems not valid anymore)

Reply:
> Not a big fan of the negated identifier. Why not define it as `enum Determinism { Enforced, Relaxed }` and use `matches!(determinism, Determinism::Relaxed)` here? Or something like that.

Renamed. Should be fine as SCALE references variants per index not per name.